### ISSUE TITLE
Default partitionField doesn't appear to work as documented

### ISSUE THREAD

**Issue by AaronDugger-IR** [GoogleCloudDataproc/spark-bigquery-connector#172]
Version: 0.15.1-beta

For the partitionField write option, the readme documentation states: 

> If not set, the table is partitioned by pseudo column, referenced via either '_PARTITIONTIME' as TIMESTAMP type, or '_PARTITIONDATE' as DATE type

I did not set the partitionField option and let the connector create a new table before writing, but the table is not partitioned. I've done tests with default save mode (error if table exists), append save mode, default intermediate format (parquet), and orc intermediate format.

**Comment by davidrabinowitz** [comment]
Have you set the `partitionType` parameter? can you share a snippet of the relevant code (replace the variable/table/bucket names of course)

**Comment by AaronDugger-IR** [comment]
No, I specifically didn't set it based on the documentation. It's not the end of the world for me, and my assumption was that if I did set it, it would need to be an actual field, so setting to _PARTITIONTIME or _PARTITIONDATE wouldn't work since those are just pseudo columns on the BigQuery side. Like I mentioned, I tried a few different setups, but this is where I landed:
```
df
  .write
  .mode(SaveMode.Append)
  .option("intermediateFormat", "orc")
  .bigquery(<fullTableName>)
```
temporaryGcsBucket is just set further up in the code directly on the SparkConf.

**Comment by davidrabinowitz** [comment]
Please try the following:
```
df
  .write
  .mode(SaveMode.Append)
  .option("partitionType", "DAY")
  .option("intermediateFormat", "orc")
  .bigquery(<fullTableName>)
```
Without adding this option, the write to BigQuery is unpartitioned.

You are right the documentation is not clear about this, we will fix it. Thanks for bringing it to our attention.

**Comment by AaronDugger-IR** [comment]
Gotcha. That worked. Though still surprising that it wasn't the default. This line seems to line up with the documentation claiming that it defaults to DAY: https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/b1177aa20863292eed3e692e4d50bcaf3c98c77c/connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala#L105

**Comment by AaronDugger-IR** [comment]
Oh, nevermind. I see now that to get to that line either partitionType or partitionField has to be defined.
